pepper-apis/src/main/java/org/broadinstitute/ddp/export/DataExporter.java

```diff
@@ -124,7 +126,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class DataExporter {
+public class DataExporter implements Closeable {
```

Comment:
Not seeing DataExporter being closed . Intentional?

Reply:
And if you do intend to close() DataExporter, then consider instantiating a DataExporter each time you `run` a Job as opposed to hanging on to the instance time after time and allow GC to reclaim whatever resources are linked back to this.
(Looking at heap dump, I think we are indirectly creating lots of threads that don't  go away. The underlying http client here might be part of this problem).

Reply:
> Not seeing DataExporter being closed . Intentional?

The DataExporter is mostly used within background jobs that exports data, so those literally hang around for the lifetime of Housekeeping. I'm implicitly letting this get closed by the JVM when Housekeeping gets destroyed.

> consider instantiating a DataExporter each time you run a Job
> ...
> I think we are indirectly creating lots of threads that don't go away

Actually, I want to do the opposite. I don't want to create new DataExporter on every `run` of a job. It's wasteful and I suspect that is causing the `indirectly creating lots of threads that don't go away` that you might have observed. Instead, we're creating the DataExporter just once (or just a few times..) and reusing that.

Reply:
OK. My concern more specifically: the elasticsearch client.
Underneath implementation it creates asynchronous apache http clients.
The worry here is that each elastic search client ends up indirectly creating a separate thread pool. We are creating more threads as we go. Heap analysis is showing me that.
So two ways to protect from building up threads unnecessarily:
1) Go with a true singleton elastic search client (Apparently that is their recommendation) 
https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/_changing_the_client_8217_s_initialization_code.html
2) Create it use and dispose of it to reduce chance of having multiple ones running at same time.

Reply:
Good find! That can be our next step in improvement! I tried digging deeper into the implementation but wasn't too familiar with the async/reactor/NIO stuff. I saw that there was a limit of "max connections" and "max connections per route", so I though there was a upper limit on that. I might be wrong there. I think I'll vote for option 1 of having a singleton.